tree/dataframe/test/dataframe_simple.cxx

```diff
@@ -896,6 +896,28 @@ TEST(RDFSimpleTests, AutomaticNamesOfHisto1DAndGraph)
 
 }
 
+TEST_P(RDFSimpleTests, DifferentTreesInDifferentThreads)
+{
+   const auto filename = "DifferentTreesInDifferentThreads.root";
+   const auto treename = "mytree";
+   {
+      auto df = RDataFrame(64)
+                   .Define("x", []() { return 1; })
+                   .Define("y", []() { return 1; })
+                   .Snapshot<int, int>(treename, filename, {"x", "y"}, {"RECREATE", ROOT::kZLIB, 4, 2, 99, false});
+   }
+
+   TFile f(filename);
+   TTree *t;
+   f.GetObject(treename, t);
+   RDataFrame df(*t);
+   *df.Define("xy", [](int x, int y) { return x * y; }, {"x", "y"})
+       .Filter([](int xy) { return xy > 0; }, {"xy"})
+       .Count();
+
+   gSystem->Unlink(filename);
+}
+
```

Comment:
this does not fail/crash in current master

Reply:
lol. I am missing the flushing every 2 entries. Coming.

Reply:
no wait. It does fail

Reply:
ok, never on my machine -- I tried both a debug and a Release build :man_shrugging:

Reply:
I take it back! Was running single-thread :sweat:

Reply:
I fell in the same trap before starting to flush the input file every 2 events (I had 1 single cluster)...